src/components/TextField/TextField.tsx

```diff
@@ -23,13 +25,51 @@ export const TextField: React.FC<Props> = ({
   placeholder = `Enter ${label}`,
   required = false,
   onChange = () => {},
+  setIsValidImg,
+  setIsValidImdb,
+
 }) => {
-  // generage a unique id once on component load
   const [id] = useState(() => `${name}-${getRandomDigits()}`);
 
-  // To show errors only if the field was touched (onBlur)
   const [touched, setTouched] = useState(false);
-  const hasError = touched && required && !value;
+  const hasError = touched && required && !value.trim();
+  let hasUrlError = false;
+
+  const pattern = new RegExp(
+    /^([A-Za-z]{3,9}:(?:\/\/)?)/.source
+    + /(?:[-;:&=+$,\w]+@)?[A-Za-z0-9.-]+|/.source
+    + /(?:www\.|[-;:&=+$,\w]+@)[A-Za-z0-9.-]+/.source
+    + /(?:\/[+~%/.\w-_]*)?\??/.source
+    + /(?:[-+=&;%@,.\w_]*)#?(?:[,.!/\\\w]*)?$/.source,
+  );
+
+  let isValidInput = pattern.test(value);
+
+  if (!isValidInput && value && (name === 'imgUrl' || name === 'imdbUrl')) {
+    hasUrlError = true;
+  }
+
+  const handleChange = (newValue: string) => {
+    isValidInput = pattern.test(newValue);
+
+    if (!isValidInput && newValue && name === 'imgUrl') {
+      setIsValidImg(false);
+    }
+
+    if (!isValidInput && newValue && name === 'imdbUrl') {
+      setIsValidImdb(false);
+    }
+
+    if (isValidInput && newValue && name === 'imgUrl') {
+      setIsValidImg(true);
+    }
+
+    if (isValidInput && newValue && name === 'imdbUrl') {
+      setIsValidImdb(true);
+    }
```

Comment:
You can remove redundant checks.
```suggestion
    if (name === 'imgUrl') {
      setIsValidImg(isValidInput);
    }

    if (name === 'imdbUrl') {
      setIsValidImdb(isValidInput);
    }
```